src/axom/core/tests/core_array.hpp

```diff
@@ -781,8 +782,9 @@ void check_device(Array<T, DIM, SPACE>& v)
   assign_raw<<<1, 1>>>(v.data(), size);
 
   // Check the contents of the array by assigning to a Dynamic array
-  // The default Umpire allocator should be Host, so we can access it from the CPU
-  Array<T, 1> check_raw_array_dynamic = v;
+  // The Umpire allocator should be Host, so we can access it from the CPU
+  int host_alloc_id = axom::getDefaultAllocatorID();
+  Array<T, 1> check_raw_array_dynamic(v, host_alloc_id);
```

Comment:
This PR does introduce a breaking change in that assigning from a device array to a dynamic array will not automatically copy data to the host; the intended replacement is to explicitly specify the host allocator ID as a constructor argument, or assign to an array explicitly-fixed as host by template parameter (see MeshTester_detail.hpp).

Reply:
Please make sure the Sphinx documentation is up to date, and discusses this.

Reply:
Thanks -- this seems much more intuitive and consistent to me -- i.e. specify a new allocator ID when you need to. Use the same one when unspecified.